slick/src/main/scala/slick/jdbc/PostgresProfile.scala

```diff
@@ -173,15 +173,13 @@ trait PostgresProfile extends JdbcProfile {
 
   class UpsertBuilder(ins: Insert) extends super.UpsertBuilder(ins) {
     override def buildInsert: InsertBuilderResult = {
-      val update = "update " + tableName + " set " + softNames.map(n => s"$n=?").mkString(",") + " where " + pkNames.map(n => s"$n=?").mkString(" and ")
-      val nonAutoIncNames = nonAutoIncSyms.map(fs => quoteIdentifier(fs.name)).mkString(",")
-      val nonAutoIncVars = nonAutoIncSyms.map(_ => "?").mkString(",")
-      val cond = pkNames.map(n => s"$n=?").mkString(" and ")
-      val insert = s"insert into $tableName ($nonAutoIncNames) select $nonAutoIncVars where not exists (select 1 from $tableName where $cond)"
-      new InsertBuilderResult(table, s"$update; $insert", ConstArray.from(softSyms ++ pkSyms))
+      val insert = s"""insert into $tableName (${allNames.mkString(",")}) values $allVars"""
```

Comment:
Isn't MERGE command more idiomatic, as part of SQL standard ? See https://wiki.postgresql.org/wiki/UPSERT

Reply:
https://www.postgresql.org/docs/9.6/static/sql-commands.html I don't see any `MERGE` syntax? I think Postgres only support `INSERT ON CONFLICT` or am I missing something?

Reply:
yep my bad, I got confused by the wiki !